src/cpp/include/openvino/genai/vision_encoder.hpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/processor_config.hpp"
+#include <openvino/openvino.hpp>
+
+namespace ov::genai {
+/// @brief A pair describing image size.
+struct HeightWidth {
+    /// @brief Height of a corresponding image.
+    size_t height;
+    /// @brief Width of a corresponding image.
+    size_t width;
+};
+
+/// @brief Embeddings of a given image. The number of slices is no
+/// greater than ProcessorConfig's max_slice_nums.
+struct EncodedImage {
+    /// @brief Embeddings of a resized image based on ProcessorConfig's
+    /// scale_resolution. The tensor's shape is
+    /// [N, H*W, hidden_size]. [N, 1014, 1152] is a possible example for
+    /// openbmb/MiniCPM-V-2. Only batch 1 is supported.
+    ov::Tensor resized_source;
+    /// @brief A size of an image used to compute embeddings for
+    /// divided by ProcessorConfig's patch_size.
+    HeightWidth resized_source_size;
+    /// @brief Embeddings of images obtained from a source image by
+    /// slicing at no more than max_slice_nums pieces and resizing.
+    /// The tensor's shape is
+    /// [slice_y, slice_x, number_of_embeddings, embedding_size].
+    /// slices_sizes.size() == slice_y * slice_x.
+    ov::Tensor slices;
+    /// @brief Flattened sizes of images used to compute embeddings
+    /// stored in slices member divided by ProcessorConfig's patch_size.
+    std::vector<HeightWidth> slices_sizes;
```

Comment:
are all slices of the same size? if no, why are they all in a single tensor `slices` ?